android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/MeFragment.kt

```diff
@@ -37,6 +31,23 @@ class MeFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_me, container, false)
     }
 
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val logoutButton : Button = view.findViewById(R.id.logout_button)
+        logoutButton.setOnClickListener {
+            // Sign out both Google account and Firebase
+            signInVM.signOut()?.addOnCompleteListener {
+                val intent = Intent(context, MainActivity::class.java)
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                startActivity(intent)
```

Comment:
addOnSuccessListener is better, thanks

As for returning a boolean status, it may return that boolean status before its completion since this is a async function.
Of course I can use while loop to wait its completion, but I am sure whether this is a good choice.